packages/grafana-data/src/field/fieldState.ts

```diff
@@ -77,7 +77,7 @@ function calculateFieldDisplayName(field: Field, frame?: DataFrame, allFrames?:
 
   // migration for dataplane with TimeSeriesMulti
   if (field.labels?.__name__ && field.labels?.__name__ === field.name) {
```

Comment:
uncomfortable that this is such a broad check that is applied to everythign... not just datsource that is migrating

Reply:
I guess the other option would be to add some property to the field, since the information needed here comes from the frame?, and the fields come from mixed frame sources?

Reply:
Moving changes into the datasource, thank you!